block_tracker.sh

```diff
@@ -286,22 +305,22 @@ fi
 cmd="execute"
 use_filter=false
 if [ $# -gt 0 ]; then
-    while true; do
+    while [[ -n "$1" ]]; do
         case "$1" in
             --disable|-d) cmd="disable" ; shift ;;
             --enable|-e) cmd="enable" ; shift ;;
             --install|-i) cmd="install" ; shift ;;
             --run|-r) cmd="install && execute" ; shift ;;
             --uninstall|-u) cmd="uninstall" ; shift ;;
             --filter|-f) use_filter=true ; shift ;;
-            "") break ;;
+   
             *)
-            write_to_console $MSG_UNKNOWN_OPTION "$1" # TBD should write help message with all accepted options
-            help
-            exit 1
-            ;;
+				write_to_console $MSG_UNKNOWN_OPTION "$1" # TBD should write help message with all accepted options
+				help
+				exit 1
+				;;
         esac
     done
 fi
 
-eval "${cmd}"
+$cmd
```

Comment:
From a semantical point of view I think it's identical. From my point of view eval should be used only when that's the only way to solve a technical programming issue. I consider eval to be harmful - not in this script but in general.